### PR TITLE
kubevirt,presubmits: update storage lanes to run with etcd in mem

### PR DESCRIPTION
**Pull request by brianmcarey**
**What this PR does / why we need it**:

The storage e2e lanes see a large number of lane failures due to etcd timeouts[1] - running these lanes with etcd in memory should stop these etcd timeouts from occurring as we have seen with the other e2e lanes.

We saw instability when running with the default tmpfs size of 512M as this was filling up and causing etcd to restart. 
```
{"level":"fatal","ts":"2024-09-18T12:16:27.580544Z","caller":"etcdserver/raft.go:224","msg":"failed to save Raft hard state and entries","error":"no space left on device","stacktrace":"go.etcd.io/etcd/server/v3/etcdserver.(*raftNode).start.func1\n\tgo.etcd.io/etcd/server/v3/etcdserver/raft.go:224"}
```

Increasing this to 1G improves the stability greatly. 

[1] https://search.ci.kubevirt.io/?search=etcdserver%3A+request+timed+out&maxAge=336h&context=1&type=build-log&name=&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by akalenyu** [comment]
@brianmcarey how is this looking? I think we're open minded about trying in mem etcd for a while to alleviate failures
@mhenriks wdyt

**Comment by brianmcarey** [comment]
> > > @brianmcarey how is this looking? I think we're open minded about trying in mem etcd for a while to alleviate failures @mhenriks wdyt
> > 
> > 
> > Not 100% sure on the stability of this - I want to run a few more rehearsals to see first. I checked the memory usage and it looks ok for the test pod with this enabled.
> 
> Correct me if I'm wrong but the test job definitions are also uncapped on the memory size (only request is specified)

Yes they do not have any memory limits set.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml [hunk 1]
```diff
@@ -1969,6 +1973,8 @@ presubmits:
           value: "true"
         - name: KUBEVIRT_WITH_ETC_IN_MEMORY
           value: "true"
+        - name: KUBEVIRT_WITH_ETC_CAPACITY
+          value: "1g"
```
Comment (akalenyu) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:1977]: hmm, not capital? 1G?

Comment (brianmcarey) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:1977]: Yes I wasn't 100% sure but according to the tmpfs man pages it should be lower case  - https://www.man7.org/linux/man-pages/man5/tmpfs.5.html

Comment (akalenyu) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:1977]: so we use the k8s unit parser to ensure correctness - https://github.com/kubevirt/kubevirtci/blob/main/cluster-provision/gocli/cmd/run.go#L305

**Review by akalenyu — APPROVED** [review]
Would it make sense to also apply this on 1.3/1.2 lanes?

**Review by dhiller — APPROVED** [review]
/approve

Thank you @brianmcarey !

Release

/hold

after the rehearsals are good.

**Comment by xpivarc** [comment]
@dhiller I think we are good to unhold

**Comment by akalenyu** [comment]
> @brianmcarey: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> rehearsal-pull-kubevirt-e2e-k8s-1.29-sig-storage-1.2	[ead44d6](https://github.com/kubevirt/project-infra/commit/ead44d6a2e3300058cff788a65b46fe70dcd7a81)	[link](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/3626/rehearsal-pull-kubevirt-e2e-k8s-1.29-sig-storage-1.2/1836699950868598784)	unknown	`/test pull-kubevirt-e2e-k8s-1.29-sig-storage-1.2`
> rehearsal-pull-kubevirt-e2e-k8s-1.28-sig-storage-1.2	[ead44d6](https://github.com/kubevirt/project-infra/commit/ead44d6a2e3300058cff788a65b46fe70dcd7a81)	[link](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/3626/rehearsal-pull-kubevirt-e2e-k8s-1.28-sig-storage-1.2/1836699951090896896)	unknown	`/test pull-kubevirt-e2e-k8s-1.28-sig-storage-1.2`

I don't think the 1.2 lanes are failing because of anything etcd or storage related. It looks like an error occurs in guest-console-log

**Comment by brianmcarey** [comment]
/hold cancel

Rehearsals look good.

**Pull request merged**